src/components/button/template.test.js

```diff
@@ -0,0 +1,254 @@
+/* globals describe, it, expect */
+
+const { render, getExamples } = require('../../../lib/jest-helpers')
+
+const examples = getExamples('button')
+
+describe('Button', () => {
+  describe('input[type=submit]', () => {
+    it('renders the default example', () => {
+      const { $ } = render('button', examples.default)
+
+      const $component = $('.govuk-c-button')
+      expect($component.get(0).tagName).toEqual('input')
+      expect($component.attr('type')).toEqual('submit')
+      expect($component.val()).toEqual('Save and continue')
+    })
+
+    it('renders with attributes', () => {
+      const { $ } = render('button', {
+        element: 'input',
+        attributes: {
+          'aria-controls': 'example-id',
+          'data-tracking-dimension': '123'
+        },
+        text: 'Save and continue'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('aria-controls')).toEqual('example-id')
+      expect($component.attr('data-tracking-dimension')).toEqual('123')
+    })
+
+    it('renders with classes', () => {
+      const { $ } = render('button', {
+        element: 'input',
+        classes: 'app-c-button--custom-modifier',
+        text: 'Save and continue'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.hasClass('app-c-button--custom-modifier')).toBeTruthy()
+    })
+
+    it('renders with disabled', () => {
+      const { $ } = render('button', examples.disabled)
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('aria-disabled')).toEqual('true')
+      expect($component.attr('disabled')).toEqual('disabled')
+      expect($component.hasClass('govuk-c-button--disabled')).toBeTruthy()
+    })
+
+    it('renders with name', () => {
+      const { $ } = render('button', {
+        element: 'input',
+        text: 'Start now',
+        name: 'start-now'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('name')).toEqual('start-now')
+    })
+
+    it('renders with type', () => {
+      const { $ } = render('button', {
+        element: 'input',
+        type: 'button',
+        text: 'Start now'
+      })
+
+      const $component = $('.govuk-c-button')
+      expect($component.attr('type')).toEqual('button')
+    })
+  })
+
+  describe('link', () => {
+    it('renders with anchor, href and an accessible role of button', () => {
+      const { $ } = render('button', examples.link)
```

Comment:
The `link` example is an implicit link, because it specifies an `href` but not an `element`. So I think this is testing the same thing as the implicit link test on [`:223`](https://github.com/alphagov/govuk-frontend/pull/461/files#diff-4411da34c01789c52a5e7379b28380d5R223)?